### DISCUSSION TITLE
How to enforce shallow copy for certain fields?

### DISCUSSION THREAD

**Discussion by eudoxos** [pydantic/pydantic#2457] [Question] [answered]
This is a MWE where I am passing around an object of type `A`,  and would like to have the object shared among two instances of `B`. I write it the naive way but the `A` is getting copied automatically:

```python
import pydantic

class A(pydantic.BaseModel):
    pass

class B(pydantic.BaseModel):
    a: A

b1=B(a=A())
b2=B(a=b1.a)

assert id(b1.a)==id(b2.a)
```

Is there a way around that?

**Comment by samuelcolvin** [comment, marked as answer]
see [Config `copy_on_model_validation`](https://pydantic-docs.helpmanual.io/usage/model_config/)

```py
from pydantic import BaseModel

class A(BaseModel):
    pass

    class Config:
        copy_on_model_validation = False


class B(BaseModel):
    a: A


b1 = B(a=A())
b2 = B(a=b1.a)

assert id(b1.a) == id(b2.a)
```

Note that:
* you need to declare `copy_on_model_validation` on the inner model, not the outer
* `copy_on_model_validation` is new in v1.8

**Reply by eudoxos** [reply]
Cool, thanks!
